src/multiplex/client.rs

```diff
@@ -138,21 +238,131 @@ where
     }
 }
 
-// ===== ClientInner =====
+// ===== Pending =====
 
-struct Pending<Tag, Item> {
+/// A pending response used to track in-flight requests.
+///
+/// Each pending response contains an associated `Tag` that is provided
+/// from the `TagStore` used to uniquely identify a request/response pair.
+pub struct Pending<Tag, Response> {
     tag: Tag,
-    tx: tokio::sync::oneshot::Sender<ClientResponse<Item>>,
+    tx: tokio::sync::oneshot::Sender<ClientResponse<Response>>,
     span: tracing::Span,
 }
 
+impl<Tag, Response> Pending<Tag, Response> {
+    /// Get the tag associated with this pending request.
+    pub fn tag(&self) -> &Tag {
+        &self.tag
+    }
+
+    /// Get the span associated with this pending request.
+    pub fn span(&self) -> &tracing::Span {
```

Comment:
I want to say yes since we already don't have a feature flag for tracing support (its there by default) I think its fair to expose? though I can remove it for now and someone can add it in a non-breaking way.

Reply:
I mean, tracing support isn't currently exposed as part of the _API_. It's there implicitly I suppose, but depends on whether you consider that part of the API contract. This on the other hand would tie us directly to the major version of `tracing`.